Source/ViewModel/MLCardFormViewModel.swift

```diff
@@ -85,27 +85,41 @@ final class MLCardFormViewModel {
             let numberField = MLCardFormField(fieldProperty: CardNumberFormFieldProperty(remoteSetting: cardNumberFieldSettings, cardNumberValue: tempTextField.getValue()))
             cardFormFields?.append([numberField])
         }
-
+        
         if let nameFieldProp = remoteSettings.filter({ $0.name == MLCardFormFields.name.rawValue }).first {
-            let nameField = MLCardFormField(fieldProperty: CardNameFormFieldProperty(remoteSetting: nameFieldProp, cardNameValue: storedCardName))
-            cardFormFields?.append([nameField])
+            if nameFieldProp.autocomplete ?? true {
+                let storedNameField = MLCardFormField(fieldProperty: CardNameFormFieldProperty(remoteSetting: nameFieldProp, cardNameValue: storedCardName))
+                cardFormFields?.append([storedNameField])
+            } else {
+                let defaultNameField = MLCardFormField(fieldProperty: CardNameFormFieldProperty(remoteSetting: nameFieldProp))
+                cardFormFields?.append([defaultNameField])
+            }
         }
-
+        
         if let expirationFieldSetting = remoteSettings.filter({ $0.name == MLCardFormFields.expiration.rawValue}).first,
-            let securityFieldSetting = remoteSettings.filter({ $0.name == MLCardFormFields.securityCode.rawValue}).first,
-            let mergedSecurityFieldSetting = MLCardFormFieldSetting.createSettingForField(.securityCode, remoteSetting: securityFieldSetting, cardUI: cardUI) {
+           let securityFieldSetting = remoteSettings.filter({ $0.name == MLCardFormFields.securityCode.rawValue}).first,
+           let mergedSecurityFieldSetting = MLCardFormFieldSetting.createSettingForField(.securityCode, remoteSetting: securityFieldSetting, cardUI: cardUI) {
             cardFormFields?.append([
                 MLCardFormField(fieldProperty: CardExpirationFormFieldProperty(remoteSetting: expirationFieldSetting)),
                 MLCardFormField(fieldProperty: CardSecurityCodeFormFieldProperty(remoteSetting: mergedSecurityFieldSetting))
-                ])
+            ])
         }
-
+        
         if let remoteIdTypes = binData?.identificationTypes, remoteIdTypes.count > 0,
-            let idNumberSetting = remoteSettings.filter({ $0.name == MLCardFormFields.identificationTypeNumber.rawValue}).first {
-            cardFormFields?.append([
-                MLCardFormField(fieldProperty: IDTypeFormFieldProperty(identificationTypes: remoteIdTypes, idTypeValue: storedIDType, keyboardHeight: measuredKeyboardSize)),
-                MLCardFormField(fieldProperty: IDNumberFormFieldProperty(identificationTypes: remoteIdTypes, idTypeValue: storedIDType, remoteSetting: idNumberSetting, idNumberValue: storedIDNumber))
-                ])
+           let idNumberSetting = remoteSettings.filter({ $0.name == MLCardFormFields.identificationTypeNumber.rawValue}).first {
+            if idNumberSetting.autocomplete ?? true {
```

Comment:
Quick question here. Should the default value of autocomplete be `true`? 
as a suggestion i think we could attach the second condition to the first one, something like:
```
if let remoteIdTypes = binData?.identificationTypes, remoteIdTypes.count > 0,
    let idNumberSetting = remoteSettings.filter({ $0.name == MLCardFormFields.identificationTypeNumber.rawValue}).first, 
    idNumberSetting.autocomplete ?? true {
```

Reply:
If we attach the validation there, we should also check for the previous conditions in the "else" case as either True (or nil) and False are valid values for the Autocomplete flag and we will be switching on them from Backend depending on the Site (e.g., MLA will have autocomplete = false, rest will have autocomplete = true).

In this case, the remoteIdTypes variable and remoteIdTypes.count>0 should validate regardless of the autocomplete flag value. Makes sense?

Reply:
Oh i see ok got it, it makes sense, then it's ok as it is.